include/net/socket.h

```diff
@@ -53,6 +53,9 @@ struct zsock_addrinfo {
 	char _ai_canonname[DNS_MAX_NAME_SIZE + 1];
 };
 
+/* Note: placing zsock_ symbols in <net/zsock.h>, and making zsock
+ * the default socket provider would greatly simplify this header.
```

Comment:
Uo remind, Zephyr's socket API functions have prefix "zsock_". Then there's a separate aliasing layer for raw POSIX names. Someone offloading Zephyr socket API would naturally do that on the level of "zsock_" functions. I don't insist though (I may imagine doing that way leads to a shorter patch), just a notice that it may need to be redone later.